zapcore/field.go

```diff
@@ -160,6 +160,23 @@ func (f Field) AddTo(enc ObjectEncoder) {
 	case NamespaceType:
 		enc.OpenNamespace(f.Key)
 	case StringerType:
+		defer func() {
+			if r := recover(); r != nil {
+				var msg string
+
+				switch v := r.(type) {
```

Comment:
should this behaviour match `fmt`?

Reply:
- `fmt` panics when printing the panic value panics. I added a test case for this and currently the code would print out `<unknown>` 
- `fmt` prints `<nil>` when the receiver is `nil` which is something we could do as well
- `fmt` prints something like `%s!(PANIC=some additional text)` which is specific to the `fmt` package IMHO but it might make sense to adopt the `PANIC=` syntax

Reply:
I added the `PANIC=` prefix and removed the case of an indefinitely reentrant panic as I think it is overly constructed. 

This simplifies the code and ensures panic values are printed the same way `fmt` behaves. 

However as we remove the `<unknown>` special case, printing the panic could causes a panic itself and that would be potentially unhandled.